Add logging and ensure child wires are stopped properly

sinkingsugar opened the pull request:
- Add trace logs in SHMesh to log the stopping of scheduled and pending wires for better debugging.
- Ensure child wires in BaseRunner are stopped and set to nullptr to avoid potential issues with dangling references.


<!--
Before submitting your PR, please review the following checklist:

- [ ] **DO NOT** submit a PR without having discussed the change first in a related issue. If none exist, create one first.
- [ ] **CONSIDER** adding a unit test if your PR resolves an issue.
- [ ] **DO** keep pull requests small so they can be easily reviewed.
- [ ] **DO** make sure all unit tests pass.
- [ ] **DO** make sure not to introduce any new compiler warnings.
- [ ] **AVOID** breaking the continuous integration build.
- [ ] **AVOID** making significant changes to the overall architecture.
-->

sinkingsugar committed:
Add logging and ensure child wires are stopped properly

- Add trace logs in SHMesh to log the stopping of scheduled and pending wires for better debugging.
- Ensure child wires in BaseRunner are stopped and set to nullptr to avoid potential issues with dangling references.

sinkingsugar committed:
Update Set to respect global status of existing variables

Adjust the Set shard to inherit the global status of an existing exposed variable if applicable. Modify logic for determining global exposure and ensure consistency in variable and table type updates.

sinkingsugar committed:
Fix null case for existingExposedType

sinkingsugar committed:
Update iOS and visionOS build targets in workflow

Replace the target `cargo-shards-rust-union-rust` with `shards-cpp-union` and `shards-core` for iOS. Re-enable visionOS build with appropriate targets (`shards-cpp-union`, `shards-core`, and `shards-lib`) and add configuration for Xcode SDK `xros`.

sinkingsugar committed:
disable vision target again (ci machine has 1.2 only)

guusw commented on shards/modules/core/core.hpp:
```diff
@@ -1220,13 +1220,18 @@ struct Set : public SetUpdateBase {
                                        existingExposedType ? &existingExposedType->exposedType : nullptr);
 
-      if (_global) {
+      if (global) {
```
Need to change Push as well, it just exposes based on _global (param)

guusw commented on shards/modules/core/wires.hpp:
```diff
@@ -191,6 +191,11 @@ struct BaseRunner : public WireBase {
         // but avoid stopping if detached and scheduled
+        if (wire->childWire) {
+          // stop the child wire if any
+          shards::stop(wire->childWire);
```
Maybe better inside stop, because this is "flow" state kind of.
Check other stop usages if it makes sense to have there

sinkingsugar committed:
Update `Push` to handle global inheritance and improve clarity

- Adjust `updateSeqInfo` and `updateTableInfo` to consider global inheritance for exposed variables using the `global` flag.
- Simplify logic by removing redundant variable type checks and ensuring consistent handling of `global` across contexts.
- Refactor code for better readability and maintainability.

sinkingsugar committed:
Refactor: improve wire ticking and child wire handling

Refactor `SHWire` and `SHMesh` to enhance wire ticking logic by introducing a `tickingWire` method and clarifying its use in step mode and child wire assignment. Adjust `stop` logic to handle child wires directly and avoid redundant stopping when detached. Clean up `BaseRunner` to remove duplicate child wire stopping logic. Simplify scheduling and state management for better coroutine handling.
